app/scripts/migrations/137.ts

```diff
@@ -0,0 +1,318 @@
+    global.sentry?.captureException(
+      new Error(
+        `Migration ${version}: typeof state.NetworkController.selectedNetworkClientId is ${typeof selectedNetworkClientId}`,
+      ),
+    );
+    return state;
+  }
+  if (!isObject(networkConfigurationsByChainId)) {
+    global.sentry?.captureException(
+      new Error(
+        `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId is ${typeof networkConfigurationsByChainId}`,
+      ),
+    );
+    return state;
+  }
+  if (!isObject(domains)) {
+    global.sentry?.captureException(
+      new Error(
+        `Migration ${version}: typeof state.SelectedNetworkController.domains is ${typeof domains}`,
+      ),
+    );
+    return state;
+  }
+
+  const getChainIdForNetworkClientId = (
+    networkClientId: string,
+    propertyName: string,
+  ): string | undefined => {
+    for (const [chainId, networkConfiguration] of Object.entries(
+      networkConfigurationsByChainId,
+    )) {
+      if (!isObject(networkConfiguration)) {
+        global.sentry?.captureException(
+          new Error(
+            `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId["${chainId}"] is ${typeof networkConfiguration}`,
+          ),
+        );
+        return undefined;
+      }
+      if (!Array.isArray(networkConfiguration.rpcEndpoints)) {
+        global.sentry?.captureException(
+          new Error(
+            `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId["${chainId}"].rpcEndpoints is ${typeof networkConfiguration.rpcEndpoints}`,
+          ),
+        );
+        return undefined;
+      }
+      for (const rpcEndpoint of networkConfiguration.rpcEndpoints) {
+        if (!isObject(rpcEndpoint)) {
+          global.sentry?.captureException(
+            new Error(
+              `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId["${chainId}"].rpcEndpoints[] is ${typeof rpcEndpoint}`,
+            ),
+          );
+          return undefined;
+        }
+        if (rpcEndpoint.networkClientId === networkClientId) {
+          return chainId;
+        }
+      }
+    }
+
+    const builtInChainId = BUILT_IN_NETWORKS.get(networkClientId);
+    if (!builtInChainId) {
+      global.sentry?.captureException(
+        new Error(
+          `Migration ${version}: No chainId found for ${propertyName} "${networkClientId}"`,
+        ),
+      );
+    }
+    return builtInChainId;
+  };
+
+  const currentChainId = getChainIdForNetworkClientId(
+    selectedNetworkClientId,
+    'selectedNetworkClientId',
+  );
+  if (!currentChainId) {
+    return state;
+  }
+
+  for (const [origin, subject] of Object.entries(subjects)) {
+    if (!isObject(subject)) {
+      global.sentry?.captureException(
+        new Error(
+          `Migration ${version}: Invalid subject for origin "${origin}" of type ${typeof subject}`,
+        ),
+      );
+      return state;
```

Comment:
Perhaps @Gudahtt can chime in here, but seems dangerous to bail out of the migration with a potentially partially modified state.

Reply:
pushing a PR with suggested change

Reply:
This doesn't quite address the problem. State mutations are still being made in the loop, resulting in partially modified state if we bail.

I'd recommend either doing validation separately in its own loop, or making changes on a cloned copy of state that isn't applied until the end

Reply:
🤔 Ok so the solution I pushed prevents the loop from bailing **internally** with partially modified state. The case you are now raising is if an external issues causes the process to bail with partially modified state. Is that correct?

Reply:
>the solution I pushed prevents the loop from bailing internally with partially modified state

It doesn't seem to 🤔

There is a loop over subjects. If the 2nd iteration encounters an error, the modifications made in the first loop are still returned.